etcdserver/api/v3rpc/watch.go

```diff
@@ -17,17 +17,20 @@ package v3rpc
 import (
 	"io"
 
+	"github.com/coreos/etcd/etcdserver"
 	pb "github.com/coreos/etcd/etcdserver/etcdserverpb"
 	"github.com/coreos/etcd/storage"
 	"github.com/coreos/etcd/storage/storagepb"
 )
 
 type watchServer struct {
+	server    etcdserver.Server
+	raftTimer *etcdserver.EtcdServer
 	watchable storage.Watchable
 }
 
-func NewWatchServer(w storage.Watchable) pb.WatchServer {
-	return &watchServer{w}
+func NewWatchServer(s etcdserver.Server, r *etcdserver.EtcdServer, w storage.Watchable) pb.WatchServer {
```

Comment:
NewWatcherServer(s *etcdserver.EtcdServer). Then you can get watchable from that arg.